pcapng/structs.py

```diff
@@ -470,42 +699,85 @@ class Options(Mapping):
         Required in order to load numeric fields.
     """
 
+    __slots__ = [
+        "schema",
+        "_field_names",
+        "data",
+        "endianness",
+    ]
+
     def __init__(self, schema, data, endianness):
-        self.schema = {}  # Schema of option fields: {<code>: {..def..}}
+        self.schema = {}  # Schema of option fields: {<code>: Option(...)}
         self._field_names = {}  # Map names to codes
-        self.raw_data = {}  # List of (code, value) tuples
+        self.data = {}  # List of (code, value) tuples
```

Comment:
I have a feeling this changed type long ago and is not a List anymore :)

Reply:
Also it sounds like this might be better suited to be a `collections.defaultdict(list)`

Reply:
I can remove the comment easily enough. I'm not following why a `defaultdict` would be preferable, could you please clarify?

Reply:
I see a number of places that checks `if not key in self.data: self.data[key] = []` before appending to `self.data[key]`.

Those are usually better simplified with a `defaultdict(list)` which is always the empty list.

As far as I can tell, that shouldn't break semantics.